linuxrc_hostname.md

```diff
@@ -1,30 +1,35 @@
 # hostname setting in installation/rescue system
 
-There are 3 cases:
+The hostname can be set (in order of precedence):
 
-1. no hostname is set
+1. via `hostname` boot option
+2. via DHCP (either from HOSTNAME or FQDN)
+3. Default hostname `install`
 
-2. hostname is set via dhcp (either from HOSTNAME or FQDN)
+linuxrc defaults to `install`, which may be overridden by DHCP(2) or user(1).
+(1) takes precedence over (2).
 
-3. hostname is set via `hostname` boot option
-
-linuxrc defaults to `install`, which may be overridden by dhcp (2) or user
-(3). (3) takes precedence over (2).
 
 ## installation system
 
-When yast is started, the value from (3) is passed to yast via
-`/etc/install.inf::Hostname` and ends up in `/etc/hostname` of the installed system.
+When yast is started, the value from (1) is passed to yast via
+`/etc/install.inf::Hostname` and ends up in `/etc/hostname` of the installed
+system.
+
+When linuxrc doesn't set `/etc/install.inf::Hostname`, yast generates a default
+`/etc/hostname` entry:
+- Before SLE12, of the form `linux-XXXX`
+- Since SLE12, `linux-XXXX.suse`
+- Since SLE15-SP2, `linux`
+`XXXX` is a random-generated alphanumerical string.
 
-When linuxrc doesn't set `/etc/install.inf::Hostname` yast generates a
-default `/etc/hostname` entry of the form `linux-XXXX` (+ `.suse` in sle12).
-`XXXX` is some random part.
+linuxrc doesn't generate the `/etc/hostname` file in the installation
+environment. Yast generates it.
```

Comment:
Really? I don't see one.

Reply:
> Really? I don't see one.

Ok, it sounds like Yast always created. I wanted to express that if it is present, it is created by Yast.

That happens when linuxrc::hostname is used, right?

Reply:
> Ok, it sounds like Yast always creates it

No! Nobody creates it. The document said

> There is no `/etc/hostname` file in the installation environment.

And that is correct (it's an empty file, to precise).

Reply:
/etc/hostname is a filename, so it is a file
empty file and no file are different things.

Reply:
An empty /etc/hostname and a missing one are semantically equivalent, afaik.

But my point was that yast does not create it. The emtpy file is always there and comes directly from the netcfg package.

If you want to push your point and stress the difference between empty and missing /etc/hostname feel free to do so, it's developer documentation after all.

But I don't want the doc to end up less accurate than before.

Reply:
empty file is incorrect and startup scripts handle it as an exception - "do not set hostname". If you try to load empty /etc/hostname in running system (e.g. using hostname -F) you'll find that it behaves differently - reports invalid hostname (for SLE-15*, for having more fun in SLE-12* it works differently - it unsets the hostname there).